diff --git a/modules/juce_gui_basics/native/juce_Windowing_mac.mm b/modules/juce_gui_basics/native/juce_Windowing_mac.mm
index ad459d483e..31209f23d6 100644
--- a/modules/juce_gui_basics/native/juce_Windowing_mac.mm
+++ b/modules/juce_gui_basics/native/juce_Windowing_mac.mm
@@ -516,57 +516,6 @@ static void releaseImageAfterDrawing()
     }
 }
 
-static Image createNSWindowSnapshot (NSWindow* nsWindow)
-{
-    JUCE_AUTORELEASEPOOL
-    {
-        // CGWindowListCreateImage is replaced by functions in the ScreenCaptureKit framework, but
-        // that framework is only available from macOS 12.3 onwards.
-        // A suitable @available check should be added once the minimum build OS is 12.3 or greater,
-        // so that ScreenCaptureKit can be weak-linked.
-       #if defined (MAC_OS_VERSION_14_0) && MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_VERSION_14_0
-        JUCE_BEGIN_IGNORE_WARNINGS_GCC_LIKE ("-Wdeprecated-declarations")
-        #define JUCE_DEPRECATION_IGNORED 1
-       #endif
-
-        CGImageRef screenShot = CGWindowListCreateImage (CGRectNull,
-                                                         kCGWindowListOptionIncludingWindow,
-                                                         (CGWindowID) [nsWindow windowNumber],
-                                                         kCGWindowImageBoundsIgnoreFraming);
-
-       #if JUCE_DEPRECATION_IGNORED
-        JUCE_END_IGNORE_WARNINGS_GCC_LIKE
-        #undef JUCE_DEPRECATION_IGNORED
-       #endif
-
-        NSBitmapImageRep* bitmapRep = [[NSBitmapImageRep alloc] initWithCGImage: screenShot];
-
-        Image result (Image::ARGB, (int) [bitmapRep size].width, (int) [bitmapRep size].height, true);
-
-        selectImageForDrawing (result);
-        [bitmapRep drawAtPoint: NSMakePoint (0, 0)];
-        releaseImageAfterDrawing();
-
-        [bitmapRep release];
-        CGImageRelease (screenShot);
-
-        return result;
-    }
-}
-
-Image createSnapshotOfNativeWindow (void* nativeWindowHandle)
-{
-    if (id windowOrView = (id) nativeWindowHandle)
-    {
-        if ([windowOrView isKindOfClass: [NSWindow class]])
-            return createNSWindowSnapshot ((NSWindow*) windowOrView);
-
-        if ([windowOrView isKindOfClass: [NSView class]])
-            return createNSWindowSnapshot ([(NSView*) windowOrView window]);
-    }
-
-    return {};
-}
 
 //==============================================================================
 void SystemClipboard::copyTextToClipboard (const String& text)
